lib.rs

```diff
@@ -158,6 +158,63 @@ mod chocolate {
             }
         }
 
+        /// Return list of projects (AccountId, ProjectId, project details(all fields))
+        #[ink(message)]
+        pub fn get_projects_list(&self) -> Vec<(AccountId, u32, Project)> {
+            let mut projects_list: Vec<(AccountId, u32, Project)> = Vec::new();
+            for id in 0..self.project_index {
+                let project = self.projects.get(id).expect("Valid keys should have project entries");
+                projects_list.push((project.owner, id, project));
+            }
+            projects_list
+
+        }
+
+        /// Return all reviews associated with a project given the project id.
+        #[ink(message)]
+        pub fn get_reviews_list(&self, project_id: u32) -> Vec<Review> {
+            self.reviews_projects_list
+                .iter()
+                .filter(|s| s.1.eq(&project_id))
+                .map(|s| {
+                    let as_key: u32 = self
+                        .reviews_projects_list
```

Comment:
You can use the enumerate method: https://stackoverflow.com/a/28991157/16071410

To get the keys of the list before the filter, that way we don't need to find it in the map

Reply:
Done!